src/main/scala/com/fulcrumgenomics/umi/ExtractUmisFromBam.scala

```diff
@@ -204,26 +225,26 @@ object ExtractUmisFromBam {
   val UmiDelimiter = "-"
 
   /**
-    * Extracts bases associated with molecular barcodes and adds them as SAM tags.  The read's bases will only contain
+    * Extracts bases associated with molecular indices and adds them as SAM tags.  The read's bases will only contain
     * template bases as defined in the given read structure.
     */
   private[umi] def annotateRecord(record: SAMRecord,
                                   readStructure: ReadStructure,
-                                  molecularBarcodeTags: Seq[String],
+                                  molecularIndexTags: Seq[String],
                                   clippingAttribute: Option[String] = None): String = {
     val bases = record.getReadString
     val qualities = record.getBaseQualityString
     // get the bases associated with each segment
     val readStructureBases = readStructure.structureReadWithQualities(bases, qualities, strict = false)
-    // get the molecular barcode segments
-    val molecularBarcodeBases = readStructureBases.collect { case SubRead(b: String, _, m: MolecularBarcode) => b }
-    // set the barcode tags
-    molecularBarcodeTags match {
-      case Seq(tag) => record.setAttribute(tag, molecularBarcodeBases.mkString(UmiDelimiter))
+    // get the molecular index segments
+    val molecularIndexBases = readStructureBases.collect { case SubRead(b: String, _, m: MolecularBarcode) => b }
+    // set the index tags
+    molecularIndexTags match {
+      case Seq(tag) => record.setAttribute(tag, molecularIndexBases.mkString(UmiDelimiter))
```

Comment:
Sure.  Though I think we may still want it....  Without it there's no way to get all the values contatenated in a single tag _without_ also writing per-UMI tags.  The latest version supports three modes:
1) Just write per umi tags by providing the appropriate number of `molecular-index-tag`s
2) Just write a single tag with all values by providing a single `molecular-index-tag`
3) Write per-umi tags _and_ a concatenated tag by providing the appropriate number of `molecular-index-tag`s _and_ `single-tag`.